code/array/array-reject-vs-delete-vs-delete_if-vs-subtract-vs-compact.rb

```diff
@@ -0,0 +1,33 @@
+require 'benchmark/ips'
+
+ARRAY = [nil]*1000
+
+def compact
+  ARRAY.compact
+end
+
+
+def reject
+  ARRAY.reject { |o| o.nil? }
```

Comment:
@ixti looks good to me use that syntax, will take on mind when made the changes proposed.

I didn't think on the mutation of the array, that was a good lesson to keep in mind.